polling_stations/apps/data_importers/management/commands/import_eoni_from_s3.py

```diff
@@ -0,0 +1,415 @@
+        )
+
+    def reproject_coord(self, x, y, transformer, row_id):
+        try:
+            return transformer.transform(x, y)
+        except Exception as e:
+            self.stdout.write(
+                self.style.WARNING(
+                    f"Warning: Failed to transform coords in {row_id} "
+                    f"{x =}"
+                    f"{y =}"
+                )
+            )
+            self.stdout.write(self.style.WARNNG(f"Exception was: {str(e)}"))
+            return None, None
+
+    def reproject_row(self, row, scheme_suffix, transformer, row_num):
+        station_x_coord_field = f"{self.station_x_field_prefix}{scheme_suffix}"
+        station_y_coord_field = f"{self.station_y_field_prefix}{scheme_suffix}"
+        new_row = {field: row[field] for field in self.address_non_coord_fields}
+        address_lon, address_lat = self.reproject_coord(
+            row[self.address_x_field],
+            row[self.address_y_field],
+            transformer,
+            row_num,
+        )
+
+        station_lon, station_lat = self.reproject_coord(
+            row[station_x_coord_field],
+            row[station_y_coord_field],
+            transformer,
+            row_num,
+        )
+
+        if not all((address_lon, address_lat, station_lon, station_lat)):
+            return None
+
+        new_row[self.address_x_field.replace("COR", "4326")] = address_lon
+        new_row[self.address_y_field.replace("COR", "4326")] = address_lat
+
+        new_row.update(
+            {
+                field: row[f"{field}{scheme_suffix}"]
+                for field in self.station_non_coord_fields
+            }
+        )
+        new_row[self.station_x_field_prefix.replace("COR", "4326")] = station_lon
+        new_row[self.station_y_field_prefix.replace("COR", "4326")] = station_lat
+        return new_row
+
+    def create_reprojected_file(self):
+        scheme_suffix = self.national_suffix
+
+        if self.import_scheme == "LOCAL":
+            scheme_suffix = self.local_suffix
+
+        transformer = Transformer.from_crs("epsg:29902", "epsg:4326", always_xy=True)
+
+        new_rows = []
+
+        row_count = 0
+        for row_num, row in enumerate(self.reader, 1):
+            row_count += 1
+            new_row = self.reproject_row(row, scheme_suffix, transformer, row_count)
+            if new_row:
+                new_rows.append(new_row)
+
+        with open(self.output_path, mode="w", newline="") as csv_file:
+            fieldnames = new_rows[0].keys()
+            writer = csv.DictWriter(csv_file, fieldnames=fieldnames)
+            writer.writeheader()
+            writer.writerows(new_rows)
+
+        self.stdout.write(
+            self.style.SUCCESS(
+                f"Successfully created reprojected file with {len(new_rows)} rows"
+            )
+        )
+        if len(new_rows) != row_count:
+            self.stdout.write(self.style.WARNING(""))
```

Comment:
Not raising here means we could just fail to reproject any rows, and we would only comment on it, rather than blowing up. 
Do we want to `raise CommandError("Didn't reproject enough rows.")` if a threshold isn't reached? 
What should that threshold be?

Reply:
OK. First off, lets not make the message here empty string. Lets write something.

Beyond that, I'll answer your question with some follow up questions:

1. Where do these logs go? If we're writing this stuff out to a log file that just goes into CloudWatch if anyone cares to retrieve it, this is not enough visibility on this problem IMO. Maybe one thing we could do is include the output from this script in the stuff we log to slack? Tbh there are several places where we are writing warnings we might want more visibility on.

2. Do we have a vibe for how often this occurs or might occur? If we think this is something we are likely to encounter often and that's OK then I guess we need to set some threshold on it and accept some number. If it is something that could happen but we don't really _expect_ to see it I would actually be inclined to just throw here even if there is only one tbh.

Reply:
So I don't think it's something we have encountered. So maybe for now raising an exception if it happens at all is the best thing. This is done in 330bed0b9

Reply:
Logging to slack at the moment is handed off to the `import_eoni` command, but there's no reason this command couldn't also do some logging if things never get as far as calling that command.